lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -681,6 +678,42 @@ protected function prepareUpdateData($entity)
         return $result;
     }
 
+    /**
+     * Decides if an extra update is required for the entity being persisted
+     * This is only required if the associated entity is different from the current one,
+     * and if the current entity relies on the database to generate its id
+     *
+     * @param mixed $entity
+     * @param mixed $newVal
+     *
+     * @return bool Returns true is an extra update is required
+     */
+    private function isExtraUpdateRequired($entity, $newVal) : bool
```

Comment:
I might be wrong, but both arguments are objects, right?
```suggestion
    private function isExtraUpdateRequired(object $entity, object $newVal) : bool
```

Reply:
@greg0ire `$newVal` might be `null` too.

This type hint is not used in the `prepareUpdateData()` method and it makes PHP7.1 tests fail, so I'll leave it without the type hint

Reply:
Oh right, your PR is on 7.1, nevermind then. Please document both parameters, but with `object` rather than `mixed`